test/bigdecimal/test_bigmath.rb

```diff
@@ -53,6 +53,17 @@ def test_cos
     assert_in_delta(0.0, cos(PI(N) * BigDecimal("301.5"), N))
   end
 
+  def test_tan
+    assert_in_delta(0.0, tan(BigDecimal("0.0"), N))
+    assert_in_delta(0.0, tan(PI(N), N))
+    assert_in_delta(1.0, tan(PI(N) / 4, N))
+    assert_in_delta(sqrt(BigDecimal(3), N), tan(PI(N) / 3, N))
+    assert_in_delta(sqrt(BigDecimal(3), 10 * N), tan(PI(10 * N) / 3, 10 * N))
+    assert_in_delta(0.0, tan(-PI(N), N))
+    assert_in_delta(-1.0, tan(-PI(N) / 4, N))
+    assert_in_delta(-sqrt(BigDecimal(3), N), tan(-PI(N) / 3, N))
```

Comment:
We need to specify an appropriate BigDecimal for the `delta` parameter in each assertion.
I'll rewrite them later.